src/webrtc/connection.ts

```diff
@@ -206,104 +221,106 @@ class WebRTCConnection implements MultiaddrConnection {
     this.verbose(`FLOW: webrtc sink 1`)
 
     // First: use relayed connection
-    await new Promise<void>((resolve) =>
-      this.relayConn.sink(
-        eagerIterator(
-          async function* (this: WebRTCConnection): Stream['source'] {
-            let webRTCFinished = false
-
-            let result: SinkType
-
-            const next = () => {
-              sourcePromise = (source as Stream['source']).next()
-              result = undefined
-            }
+    await new Promise<void>((resolve, reject) =>
```

Comment:
Please document this block of logic more. The way its implemented is fairly complex.

Reply:
added a few comments